python/cuda_cccl/cuda/cccl/cooperative/experimental/block/_block_histogram.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+cuda.cccl.cooperative.block.histogram
+=====================================
+
+This module provides a set of :ref:`collective <collective-primitives>`
+methods for constructing block-wide histograms from data samples partitioned
+across CUDA thread blocks.
+
+It contains an implementation of the "one-shot" `cub::BlockHistogram`
+primitive.
+"""
+
+from enum import IntEnum, auto
+from typing import Callable
+
+import numba
+
+from .._common import (
+    make_binary_tempfile,
+    normalize_dim_param,
+    normalize_dtype_param,
+)
+from .._types import (
+    Algorithm,
+    Dependency,
+    DependentArray,
+    Invocable,
+    Pointer,
+    TemplateParameter,
+)
+from .._typing import (
+    DimType,
+    DtypeType,
+)
+
+
+class BlockHistogramAlgorithm(IntEnum):
+    """
+    Enum representing the different algorithms available for block histogram.
+    """
+
+    SORT = auto()
+    ATOMIC = auto()
+
+    def __str__(self):
+        return f"::cub::BLOCK_HISTO_{self.name.upper()}"
+
+
+def histogram(
+    item_dtype: DtypeType,
+    counter_dtype: DtypeType,
+    dim: DimType,
+    items_per_thread: int,
+    bins: int,
+    algorithm: BlockHistogramAlgorithm = None,
+) -> Callable:
+    """
+    Creates a block-wide histogram primitive based on the CUB library's
+    BlockHistogram functionality.
+
+    :param item_dtype: Supplies the data type of the items being sampled.
```

Comment:
suggestion: consider adding literalinclude to link code from the test as snippet here